### ISSUE TITLE
Destructuring 'prop-types' results in "no-typos" False Positive

### ISSUE THREAD

**Issue by r-mc2** [yannickcr/eslint-plugin-react#2136]
(I took a look at other open issues here and found some related but none pointing at exactly this)

I prefer to destructure the exact types I need from the `prop-types` package rather than pull the whole thing in and dot-syntax to what I need. This seems to be a missing case in the rule file where it only makes [two assumptions](https://github.com/yannickcr/eslint-plugin-react/blob/84652b6527161f651a671754fb5f619296c3654e/lib/rules/no-typos.js#L156):
* Importing from the new package: `import PropTypes from 'prop-types';`
* Importing from the old react package: `import { PropTypes } from 'react';`

In my case I go with:
```jsx
import { string, element } from "prop-types";

class Sample {
 ...
}

Sample.propTypes = {
  title: string.isRequired,
  body: element.isRequired
};
```

Both of the above cause a "`Typo in declared prop-type: isRequired`". Which in digging through the code makes sense as it appears that because I do not use `PropTypes.string.isRequired` it is looking for `isRequired` inside the known types list in the `prop-types`. And when not found in its iteration the test fails and I get the false-positive.

I'm not entirely sure how to add support for this use-case of `prop-types` but I'm pretty sure it's this [if-block](https://github.com/yannickcr/eslint-plugin-react/blob/84652b6527161f651a671754fb5f619296c3654e/lib/rules/no-typos.js#L98) that would also need some work as this is where the logic is falling through to test `string.isRequired` with the same check responsible for testing `PropTypes.string`.

**Comment by ljharb** [comment]
I've added passing test cases to master. If you can provide more info such that I can reproduce the failure, I'll be happy to reopen this - otherwise it might be fixed on master but unreleased.

**Comment by r-mc2** [comment]
Thanks for looking into this.

Threw together a [quick gist](https://gist.github.com/r-mc2/0d62ef180ec6be8163a07782c32e948f) to demonstrate better what I'm working with when the `no-typos` pops up.

And a screenshot from my editor:
![screen shot 2019-01-17 at 12 13 21 pm](https://user-images.githubusercontent.com/36780233/51336095-be5dc400-1a51-11e9-9e25-1d9566eb8d25.png)

Also should have included my version number for you last time, apologies:
`"eslint-plugin-react": "7.4.0",`

**Comment by ljharb** [comment]
Ah, well that’s ancient :-) can you update it from 7.4 to 7.12.4 and try again?

**Comment by r-mc2** [comment]
Hey, sorry for the delay in getting back to you, but yes updating to latest (7.12.4) did resolve the false-positive. Thanks, Jordan!
